src/app/modules/entry-live/views/live-discovery-chart/live-discovery-request-factory.ts

```diff
@@ -34,43 +46,72 @@ export class LiveDiscoveryRequestFactory implements RequestFactory<KalturaMultiR
 
   private _getGraphActionArgs: ReportGetGraphsActionArgs = {
     reportType: liveReportTypeMap(KalturaReportType.discoveryRealtime),
-    reportInputFilter: new KalturaReportInputFilter({
+    reportInputFilter: new KalturaEndUserReportInputFilter({
       timeZoneOffset: DateFilterUtils.getTimeZoneOffset(),
       toDate: this._dateRange.toDate,
       fromDate: this._dateRange.fromDate,
       interval: this._interval,
     }),
     responseOptions: this._responseOptions
   };
-  
+
   public set dateRange(range: DateRangeServerValue) {
     if (range.hasOwnProperty('toDate') && range.hasOwnProperty('fromDate')) {
       this._dateRange = range;
       this.onPollTickSuccess();
     }
   }
-  
-  constructor(private _entryId: string) {
-    this._getGraphActionArgs.reportInputFilter.entryIdIn = this._entryId;
-    this._getTotalActionArgs.reportInputFilter.entryIdIn = this._entryId;
+
+  constructor(private activationArgs: WidgetsActivationArgs) {
+    this._getGraphActionArgs.reportInputFilter.entryIdIn = this.activationArgs.entryId;
```

Comment:
@viktor-kyryltsev-kaltura we can replace the entire code below with a shorter version:
```
    ['entryIdIn','countryIn','regionIn','citiesIn','deviceIn','operatingSystemIn','browserIn','userIds'].forEach(filter => {
      if (this.activationArgs[filter]) {
        (this._getGraphActionArgs.reportInputFilter as KalturaEndUserReportInputFilter)[filter] = this.activationArgs[filter];
        (this._getTotalActionArgs.reportInputFilter as KalturaEndUserReportInputFilter)[filter] = this.activationArgs[filter];
      }
    });
```
this shortcut can be used in all places below

Reply:
changed in all places

Reply:
I double checked and filters are applying to discovery chart